snips_nlu/pipeline/configs/config.py

```diff
@@ -20,9 +18,30 @@ class ProcessingUnitConfig(with_metaclass(ABCMeta, Config)):
     """Represents the configuration object needed to initialize a
         :class:`.ProcessingUnit`"""
 
-    @classproperty
+    @abstractproperty
     def unit_name(self):
         raise NotImplementedError
 
+    def set_unit_name(self, value):
```

Comment:
Why not something like: 
```python
@unit_name.setter
def unit_name(self, value):
    pass
```

Reply:
It's not possible in the current state because `unit_name` is defined as an abstract property in the `ProcessingUnitConfig` class, and abstract properties defined with the `@abstractpropery` decorator are _read only_.